tests/testthat/_snaps/external-generic.md

```diff
@@ -5,3 +5,41 @@
     Output
       <S7_external_generic> foo::bar(x)
 
+# new_method works with both hard and soft dependencies
+
+    Code
+      args(Foo)
+    Output
+      function (bar = t0::AnS7Class()) 
+      NULL
+    Code
+      args(t2::AnS7Class2)
+    Output
+      function (bar = t0::AnS7Class()) 
```

Comment:
Random thought — should probably check with a class with a non-syntactic name too.